### ISSUE TITLE
some midi objcts crash Pd 0.54.1 - Mac Book M1 Pro (6) 32GB 

### ISSUE THREAD

**Issue by seppog** [porres/pd-else#1988]
midi.in crashes pd 0.54.1 (with no midi devices in settings, and with midi device) Mac Book M1 Pro (6) 32GB
OS Ventura 13.6.2 (22G320)

**Comment by porres** [comment]
how exactly does it crash?

**Comment by porres** [comment]
does [midiin] from vanilla crash?

nothing crashes here, so it's hard to see what's wrong and fix it

**Comment by seppog** [comment]
the following else midi objects crash pd vanilla 0.54.1 on M1 (Ventura) when creating them (pd just closes without any error or other message. also OS-X does not show any messages or errors):

midi.learn, midi.in, midi.out, ptouch.in, pgm.in. 

All other objects seem to work, at least loading All_objects.pd loads when eliminating the above objects/abstractions.

**Comment by seppog** [comment]
p.s. also crashes the same way on M1 Air from two students (don't know their os version)

**Comment by porres** [comment]
does [midiin] from vanilla crash?

**Comment by seppog** [comment]
no just all vanilla midi from the midi-help.pd file  works

**Comment by porres** [comment]
Ok, so only 

**- [midi.learn] 
- [midi.in] 
- [midi.out] 
- [ptouch.in] 
- [pgm.in]**

fucking weird... like, why does [ptouch.in] and not [touch.in]? and why not [ptouch.out]? It's weird that some MIDI objects from the library work...

Oh boy...

**Comment by seppog** [comment]
did check again - 
<img width="301" alt="Bildschirmfoto 2023-12-11 um 01 08 11" src="https://github.com/porres/pd-else/assets/23729035/37cf468e-9490-47c6-8aae-d6139e838f96">

hast strange - removed the objects/externals from the /else directory so they can't be created

**Comment by seppog** [comment]
ok here is the log file from the crash, hope it helps
[pd-2023-12-11-002015.txt](https://github.com/porres/pd-else/files/13629756/pd-2023-12-11-002015.txt)

**Comment by porres** [comment]
hmmm, you know what, what if you use plugdata, does it crash?

maybe @timothyschoen can help us and also make sense of the log file

**Comment by porres** [comment]
> that's really strange - removed the objects/externals from the /else directory so they can't be created

what is strange?

**Comment by porres** [comment]
and can you test earlier versions of the library and Pd?

**Comment by seppog** [comment]
a) It does not crash in plug data on the same machine.
b) It worked with pd vanilla and Else 1.0-0 rc-9 on the same machine

**Comment by porres** [comment]
> b) It worked with pd vanilla and Else 1.0-0 rc-9 on the same machine

with pd 54-1 right?

**Comment by porres** [comment]
@seppog  it is weird to me is why [midi.out] crashes but no compiled midi out object from else is broken

**Comment by martindupras** [comment]
On mac M3 Max, running MacOS Sonoma 14.2.1, PD vanilla 0.54-1:

All_objects.pd makes PD crash, as do several of the help patches that make use of pgm.in and ptouch.in. 

Removing the following files from the ELSE directory:

pgm.in.d_fat
ptouch.in.d_fat

allows All_objects.pd, albeit with the following errors in the PD console:

--- 
else/pgm.in
... couldn't create
else/ptouch.in
... couldn't create
else/pgm.in
... couldn't create
else/ptouch.in
... couldn't create
else/pgm.in
... couldn't create
else/pgm.in
... couldn't create
else/ptouch.in
... couldn't create
else/pgm.in
... couldn't create
else/ptouch.in
... couldn't create
else/ptouch.in
... couldn't create
[dust~]: improper args
else/dust~
... couldn't create
[lfnoise~]: improper args
else/lfnoise~
... couldn't create
---

**Comment by porres** [comment]
that's great and it proves all other objects don't crash. @seppog can you confirm if bend.in actually works for you, as well as [midi.out]? can you do the same test and delete the binaries of pgm.in and ptouch.in and see how it goes?

**Comment by porres** [comment]
on a separate issue, it's weird that  [dust~] and [lfnoise~] cannot be created because of "improper args".  We might open a separate issue for it but this looks way too weird to me. Let's double check, create a new blank patch and try creating these objects without args, let's see what happens. @seppog could you also test this?

**Comment by seppog** [comment]
Bend.in works,
had to delete already last time midi.in <http://midi.in/>, midi.out midi.learn touch.in pgm.in so that all objects don’t crash.

dust~ works with arguments, as does flnoise~work with an argument.

Best
Seppo

> Am 11.01.2024 um 18:02 schrieb porres ***@***.***>:
> 
> 
> that's great and it proves all other objects don't crash. @seppog can you confirm if bend.in actually works for you, as well as [midi.out]? can you do the same test and delete the binaries of pgm.in and ptouch.in and see how it goes?
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.Message ID: ***@***.***>

— 
My generation really fucked it up, ask the last one !

Josef (Seppo) Gruendler
gsm/signal +43-670-556-2019
***@***.***@fairmove.net 
telegram @gseppo
http://gruendler.mur.at
Last publication:
Sounds in Grid: History and Development of Grid-Based Musical Interfaces and their Rooting in Sound, Interaction and Screen Design: https://www.degruyter.com/document/doi/10.1515/9783110733228-010/html

Be green keep it on the screen!

**Comment by porres** [comment]
> dust~ works with arguments, as does flnoise~work with an argument.

does it mean they dont work wihtout args?

i Guess [midi.out] didn't really have a problem, as it's just an abstraction with all the real midi out objects

**Comment by seppog** [comment]
No, I can create an dust~ object without any arguments, but when loading else at startup I get the error messages [else/dust~]: improper args and [else/dust~]: ... couldn't create, same with lfnoise~

best
S.

> Am 11.01.2024 um 21:00 schrieb porres ***@***.***>:
> 
> 
> dust~ works with arguments, as does flnoise~work with an argument.
> does it mean they dont work wihtout args?
> i Guess [midi.out] didn't really have a problem, as it's just an abstraction with all the real midi out objects
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.Message ID: ***@***.***>

— 
If you could predict the growth of knowledge, your knowledge wouldn’t grow.(Sabine Hossenfelder)

Josef (Seppo) Gruendler
gsm/signal +43-670-556-2019
***@***.***@fairmove.net 
telegram @gseppo
http://gruendler.mur.at
Last publication:
Sounds in Grid: History and Development of Grid-Based Musical Interfaces and their Rooting in Sound, Interaction and Screen Design: https://www.degruyter.com/document/doi/10.1515/9783110733228-010/html

Be green keep it on the screen!

**Comment by seppog** [comment]
Yes Bose work with and without arguments, just when loading else at startup I get this error messages ( bad args etc) afterwards it works just fine .
Midi.out seems to work

> Am 11.01.2024 um 21:00 schrieb porres ***@***.***>:
> 
> 
> dust~ works with arguments, as does flnoise~work with an argument.
> does it mean they dont work wihtout args?
> i Guess [midi.out] didn't really have a problem, as it's just an abstraction with all the real midi out objects
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.Message ID: ***@***.***>

— 
If the human brain were so simple that we could understand it, then we would be so simple that we couldn't. (Emerson Pugh)

Josef (Seppo) Gruendler
gsm/signal +43-670-556-2019
***@***.***@fairmove.net 
telegram @gseppo
http://gruendler.mur.at
Last publication:
Sounds in Grid: History and Development of Grid-Based Musical Interfaces and their Rooting in Sound, Interaction and Screen Design: https://www.degruyter.com/document/doi/10.1515/9783110733228-010/html

Be green keep it on the screen!

**Comment by porres** [comment]
> No, I can create an dust~ object without any arguments, but when loading else at startup I get the error messages [else/dust~]: improper args and [else/dust~]: ... couldn't create, same with lfnoise~

sorry, that doesn't make any sense at all to me... 

do you mean that when you have 'else' at startup binaries and you open Pd (without any patches, just the binary) you get this message?

Can I have screenshots?

**Comment by porres** [comment]
anyway, I made random changes and maybe something magical happened in the last commit, you can try binaries from github actions I guess, see https://github.com/porres/pd-else/actions/runs/7494849219  let me know how it went @seppog and @martindupras 

I am requesting the help of @timothyschoen anyway as I guess he's the one who could really save us

**Comment by seppog** [comment]
An no sorry, loads without any error when loaded at startup.
Loading all objects give me the errors.
But then creating the objects runs smoothly.

Sorry for the confusion
Screenshots attached

**Comment by porres** [comment]
> Screenshots attached

cant find it

**Comment by seppog** [comment]
Strange, made a zip and attached again




> Am 12.01.2024 um 01:00 schrieb porres ***@***.***>:
> 
> 
> Screenshots attached
> cant find it
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.Message ID: ***@***.***>

— 
My generation really fucked it up, ask the last one !

Josef (Seppo) Gruendler
gsm/signal +43-670-556-2019
***@***.***@fairmove.net 
telegram @gseppo
http://gruendler.mur.at
Last publication:
Sounds in Grid: History and Development of Grid-Based Musical Interfaces and their Rooting in Sound, Interaction and Screen Design: https://www.degruyter.com/document/doi/10.1515/9783110733228-010/html

Be green keep it on the screen!

**Comment by martindupras** [comment]
> anyway, I made random changes and maybe something magical happened in the last commit, you can try binaries from github actions I guess, see https://github.com/porres/pd-else/actions/runs/7494849219 let me know how it went @seppog and @martindupras
> 

I tried the build at that link.

**ptouch.in** and **pgm.in** both load correctly.

However, I get the following errors in the console:

`[dust~]: improper args
else/dust~
... couldn't create
/Users/martindupras/Documents/Pd/externals/else/sfz~.d_arm64:dlopen(/Users/martindupras/Documents/Pd/externals/else/sfz~.d_arm64, 0x000A): tried: '/Users/martindupras/Documents/Pd/externals/else/sfz~.d_arm64' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/Users/martindupras/Documents/Pd/externals/else/sfz~.d_arm64' (no such file), '/Users/martindupras/Documents/Pd/externals/else/sfz~.d_arm64' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64'))
else/sfz~
... couldn't create
[lfnoise~]: improper args
else/lfnoise~
... couldn't create
/Users/martindupras/Documents/Pd/externals/else/circuit~.d_arm64:dlopen(/Users/martindupras/Documents/Pd/externals/else/circuit~.d_arm64, 0x000A): tried: '/Users/martindupras/Documents/Pd/externals/else/circuit~.d_arm64' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/Users/martindupras/Documents/Pd/externals/else/circuit~.d_arm64' (no such file), '/Users/martindupras/Documents/Pd/externals/else/circuit~.d_arm64' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64'))
else/circuit~
... couldn't create
else/
... couldn't create`

**Comment by porres** [comment]
> ptouch.in and pgm.in both load correctly.

I made no changes at all to ptouch.in, damn!, sfz~ and circuit~ can't be built via actions, dust~ is now another mistery we should maybe discuss in a separate issue

**Comment by porres** [comment]
> Strange, made a zip and attached again

you need to log into the web at https://github.com/porres/pd-else/issues/1988 and upload the zip

**Comment by porres** [comment]
well, I shouldn't, but I am also using this issue to talk about dust~, dust2~ and lfnoise~ now that pgm.in and ptouch.in magically work (when I made no changes at all to ptouch.in).

So, I made some rather random changes to dust~, dust2~ and lfnoise~. I am doing things a bit diferently but I am also now printing more detailed errors, so if you still get some "improper args" message now there should be a better clue on at leadt where is the problem.

You can get the binaries at https://github.com/porres/pd-else/actions/runs/7505069438

please test it folks, thanks

**Comment by seppog** [comment]
[screenshots.zip](https://github.com/porres/pd-else/files/13922129/screenshots.zip)

**Comment by porres** [comment]
>[screenshots.zip](https://github.com/porres/pd-else/files/13922129/screenshots.zip) 

I'm sorry, this is very inconsistent and hard to follow... now I see a screenshot with many objects that do not load that I hadn't heard before and that are not in the patch anyway. I can't make anything out of this. I have no idea what is going on or how to reproduce the error.

![Bildschirmfoto 2024-01-12 um 00 26 26](https://github.com/porres/pd-else/assets/5360142/429dbcdb-0066-42d5-9d85-c7ac49a1b191)

**Comment by seppog** [comment]
sorry for me causing confusion:
this happens when I loaded  the All_objects.pd file. 

to make it more complicated I loaded Your new binaries (without any deleted binaries etc) and it loads without crashing pd but with the following error messages (screenshot attached) 
But if I open the help files [rand.f~] works, but sfz~and circuit~won't 
![Bildschirmfoto 2024-01-12 um 18 31 32](https://github.com/porres/pd-else/assets/23729035/2a554587-ac75-427f-b9ab-6f627bcabd81)
![Bildschirmfoto 2024-01-12 um 18 35 28](https://github.com/porres/pd-else/assets/23729035/954c5df6-ebd0-4fef-95bf-8b2f2ef06e25)

**Comment by porres** [comment]
[sfz~] and [circuit~] should not work, it's fine. I guess it's time for me to make a pre release of RC 11, then you can download and test things (I will also provide working [sfz~] and [circuit~] binaries). It will be a proper release I'll make here on github and will also upload to deken!

**Comment by porres** [comment]
there's a new test release up in deken, see if you can find it

<img width="657" alt="Screen Shot 2024-01-12 at 16 25 28" src="https://github.com/porres/pd-else/assets/5360142/01286f9d-706e-46f6-a6c7-5272f8c5cb84">



it's also here

https://github.com/porres/pd-else/releases/tag/v.1.0-rc11-test


all binaries, including circuit~ and sfz~ should work, open all_objects.pd and let me know what happens please @martindupras and @seppog  !


it works fine here on my mac intell macbook pro from 2018 and takes about 50% of the CPU (please also test with the DSP on)

<img width="984" alt="Screen Shot 2024-01-12 at 16 29 26" src="https://github.com/porres/pd-else/assets/5360142/78a0051d-b32f-49e1-9c6e-bb0a3813ed63">

**Comment by porres** [comment]
I did not update the print message on the terminal, so be sure to delete the old folder and nuke it away... one way to be sure is checking if you have the new objects like [mtx.mc~]  in [pd Control(Fader/Panning/Routing)], the one in the bottom here

<img width="326" alt="Screen Shot 2024-01-12 at 16 32 18" src="https://github.com/porres/pd-else/assets/5360142/f758f414-60ae-4c0a-970b-3ad068e0a290">

**Comment by porres** [comment]
2 people have already reported no crash or errors on the console, which is promising...

but I need you both @seppog and @martindupras to confirm there's no issue before closing this for good, and I'll be sure to ping you again when the next update officially comes out soon!

**Comment by martindupras** [comment]
I have just downloaded the rc11 pre-test release from deken and it all seems to be working; no error messages at all. 

Awesome!

**Comment by porres** [comment]
damn, I have to keep this open, someone on facebook just got an error with rand.f~

![418446707_7872878419413484_3133651524262499034_n](https://github.com/porres/pd-else/assets/5360142/e6d8bff5-98ae-4ede-ba53-3e745d184cb9)

[Sasha Borisov](https://www.facebook.com/groups/4729684494/user/100000742112027/?__cft__[0]=AZWBAQCNgi-HBundByr9EnmY97Vb8WHLcYyt1YXQahFLwSutfgOYAGribYYyPJClzJyGw4KJj8B3HEt0WXUkgt1vRrkHqwYwrpW-_40mriyMXh7MWqTrZjenL4qNjJSgWxe2wcS-gFqaD2gM8zEiXXCU&__tn__=R]-R)
mac mini 2020 m1 macOS 13.6.3
All_objects.pd opens for the first time for about 2 minutes with running processes: syspolicyd, XprotectService. after restarting Pd it opens immediately, but always with this error:
[rand.f~]: improper args
else/rand.f~

**Comment by porres** [comment]
a new '1.0-0_rc11-Pre-Test-Release2' is up in deken and I may have fixed rand.f~ like I did fix dust~ and stuff, waiting for the user to confirm it is fine now...

**Comment by martindupras** [comment]
I just tried '1.0-0_rc11-Pre-Test-Release2' and I now get this error:

`[rand.i~]: improper args
else/rand.i~
... couldn't create
<img width="907" alt="Screenshot 2024-01-12 at 22 34 30" src="https://github.com/porres/pd-else/assets/16140857/4f8c2bf0-e7c2-4777-85a2-23cedb1acdbe">
`

**Comment by porres** [comment]
oh, c'mon.. see how fucking weird this is?

**Comment by porres** [comment]
anyway, will go ahead and fix rand.i~ which must be the same code...

**Comment by porres** [comment]
I made a search and seems this is the last bad one....

**Comment by martindupras** [comment]
I've just tried about 20 times. I get the error maybe 20% of the time. Some of the time I get the "#mouse_mouse: no such object" message. There doesn't seem to be a pattern, it seems completely random.

**Comment by porres** [comment]
"#mouse_mouse: no such object" is another known issue

**Comment by martindupras** [comment]
I've just tried systematically another 20 times. This time I kept a tally and I got the error 11 out of 20 times.

`Tally:


1: 
[rand.i~]: improper args
else/rand.i~
... couldn't create

2:
no errors

3:
[rand.i~]: improper args
else/rand.i~
... couldn't create

4:
[rand.i~]: improper args
else/rand.i~
... couldn't create

5:
no errors

6:
[rand.i~]: improper args
else/rand.i~
... couldn't create

7:
no errors

8:
[rand.i~]: improper args
else/rand.i~
... couldn't create

9:
no errors

10:
[rand.i~]: improper args
else/rand.i~
... couldn't create

11:
[rand.i~]: improper args
else/rand.i~
... couldn't create

12:
no errors

13:
[rand.i~]: improper args
else/rand.i~
... couldn't create

14:
no errors

15:
[rand.i~]: improper args
else/rand.i~
... couldn't create

16:
[rand.i~]: improper args
else/rand.i~

17:
no errors

18:
no errors

19:
[rand.i~]: improper args
else/rand.i~
... couldn't create

20:
no errors`

**Comment by porres** [comment]
'1.0-0_rc11-Pre-Test-Release3' is up

**Comment by porres** [comment]
I will go ahead and close this, but please double chceck confirm me when you can @seppog and @martindupras that all objects are loading fine

**Comment by seppog** [comment]
Installed RC-11 from Denen, all objects are loading (takes some time, approx 100sec) but without any errors. tnx a lot

**Comment by porres** [comment]
>  (takes some time, approx 100sec)

@seppog how long did it take before? Should be the same as I did nothing different

**Comment by martindupras** [comment]
> Installed RC-11 from Denen, all objects are loading (takes some time, approx 100sec) but without any errors. tnx a lot

I have the same sort of behaviour some of the time but not always and only the first time I open All_objects.pd after re-installing Else again. (I observed that in pre-test-release2 but assumed it was normal behaviour".) 

I just tried again systematically. Here's what I observed. If I delete the 'else' folder from my externals and download again from Deken, either Deken reports very quickly that installation was completed BUT some objects are still downloading in the background; or it takes a little while to download and by the time it reports that Else is installed, All_objects.pd loads

I wonder if perhaps the issue is related to the number of files (1143, from what I can see?) If  the server times out for a bit after a certain number of files, that would explain why the patch doesn't load until all the objects are there.
